src/autolua.zig

```diff
@@ -36,6 +36,7 @@ pub fn push(L: ?*lua.lua_State, value: anytype) void {
     const T = @TypeOf(value);
     switch (@typeInfo(@TypeOf(value))) {
         .Void => lua.lua_pushnil(L),
+        .Null => lua.lua_pushnil(L),
```

Comment:
I don't think mapping `null` to `nil` is the best default... many lua libraries distinguish `null` when working with external resources, e.g. [`dkjson.null`](http://dkolf.de/src/dkjson-lua.fsl/wiki?name=Documentation).

It's somewhat common to make null `lua_pushlightuserdata(L, NULL)`...

I think we may need to pass in some `comptime options: PushOptions` field to allow this to be configurable.

Reply:
It makes a lot of sense actually, but wouldn't that need to make the developer modify the `autolua.zig` file for it's configuration?

I was thinking of just making `options` a `pub var` which allowed to change `autolua.zig`'s configuration from any file that imported said library, as in:

```zig
const autolua = @import("autolua.zig")

pub fn main() void {
    autolua.options.allowNil = true
    ...
```

This would be relatively efficient and would de-complicate things a lot but I'm not sure about it

Reply:
I just implemented the after-mentioned idea, this is not final unless said so.